### PR TITLE
Expression Builder Draft

### PR DESCRIPTION
**Pull request by BlackPhoenix134**
First draft after like 4 iterations.
Would probably be best to ditch DSL and work with the internal structs.

### CODE REVIEW

File: src/cs/production/Flecs/Expression/Expression.cs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        termBuilder.Second<T2>();
+        return termBuilder;
+    }
+
+    public TermBuilder Term()
+    {
+        return new TermBuilder(_world);
+    }
+
+    public class TermBuilder
```
Comment (lithiumtoast) [inline src/cs/production/Flecs/Expression/Expression.cs]: Can we have one class/struct per file?

Comment (BlackPhoenix134) [reply src/cs/production/Flecs/Expression/Expression.cs]: Of course

Comment (BlackPhoenix134) [reply src/cs/production/Flecs/Expression/Expression.cs]: But it does not make sense, does it?
The term builder is not something you would use outside that scope, other than for chaining

Comment (lithiumtoast) [reply src/cs/production/Flecs/Expression/Expression.cs]: Create a folder then for everything in related to expressions? I see you already did that.

Comment (BlackPhoenix134) [reply src/cs/production/Flecs/Expression/Expression.cs]: Alrighty

**Review by lithiumtoast — DISMISSED** [review]

**Review by lithiumtoast — APPROVED** [review]

**Pull request merged**